### ISSUE TITLE
release: v3.6.0 issues overview 

### ISSUE THREAD

**Issue by bobleesj** [diffpy/diffpy.utils#143]
### 3.6.0 release timeline

Nov 27, 2024 - Dec 30, 2024

@sbillinge 

- [x] https://github.com/diffpy/diffpy.utils/issues/181
- [x] https://github.com/diffpy/diffpy.utils/issues/177
- [x] https://github.com/diffpy/diffpy.utils/issues/185
- [x] https://github.com/diffpy/diffpy.utils/issues/244
- [x] https://github.com/diffpy/diffpy.utils/issues/245

@bobleesj

- [x] https://github.com/diffpy/diffpy.utils/issues/182
- [x] https://github.com/diffpy/diffpy.utils/issues/188
- [x] https://github.com/diffpy/diffpy.utils/issues/194
- [x] https://github.com/diffpy/diffpy.utils/issues/201
- [x] https://github.com/diffpy/diffpy.utils/issues/200
- [x] https://github.com/diffpy/diffpy.utils/issues/212
- [x] https://github.com/diffpy/diffpy.utils/issues/213
- [x] https://github.com/diffpy/diffpy.utils/issues/224
- [x] https://github.com/diffpy/diffpy.utils/pull/221
- [x] https://github.com/diffpy/diffpy.utils/pull/225
- [x] https://github.com/diffpy/diffpy.utils/issues/224
- [x] https://github.com/diffpy/diffpy.utils/issues/227
- [x] https://github.com/diffpy/diffpy.utils/issues/223
- [x] https://github.com/diffpy/diffpy.utils/issues/240
- [x] https://github.com/diffpy/diffpy.utils/issues/257
- [x] https://github.com/diffpy/diffpy.utils/issues/258
- [x] https://github.com/diffpy/diffpy.utils/issues/187
- [x] https://github.com/diffpy/diffpy.utils/issues/241
- [x] https://github.com/diffpy/diffpy.utils/issues/239
- [x] https://github.com/diffpy/diffpy.utils/issues/7

@bobleesj and @sbillinge - establish group's testing standards
- [x] https://github.com/diffpy/diffpy.utils/pull/231
- [x] https://github.com/diffpy/diffpy.utils/pull/236
- [x] https://github.com/diffpy/diffpy.utils/pull/255

@yucongalicechen

- [x] https://github.com/diffpy/diffpy.utils/issues/49
- [x] https://github.com/diffpy/diffpy.utils/issues/50
- [x] https://github.com/diffpy/diffpy.utils/issues/173
- [x] https://github.com/diffpy/diffpy.utils/issues/174
- [x] https://github.com/diffpy/diffpy.utils/issues/175
- [x] https://github.com/diffpy/diffpy.utils/issues/176
- [x] https://github.com/diffpy/diffpy.utils/issues/186
- [x] https://github.com/diffpy/diffpy.utils/issues/183
- [x] https://github.com/diffpy/diffpy.utils/issues/226
- [x] https://github.com/diffpy/diffpy.utils/issues/238
- [x] https://github.com/diffpy/diffpy.utils/issues/247
- [x] https://github.com/diffpy/diffpy.utils/issues/256
- [x] https://github.com/diffpy/diffpy.utils/issues/281
- [x] https://github.com/diffpy/diffpy.utils/issues/274
- [x] https://github.com/diffpy/diffpy.utils/issues/256
- [x] https://github.com/diffpy/diffpy.utils/issues/287
- [x] https://github.com/diffpy/diffpy.utils/issues/45
- [x] https://github.com/diffpy/diffpy.utils/issues/248
- [x] https://github.com/diffpy/diffpy.utils/issues/287

@yucongalicechen and @bobleesj 
- [x] https://github.com/diffpy/diffpy.utils/issues/53

@alisnwu
- [x] https://github.com/diffpy/diffpy.utils/issues/210
- [x] https://github.com/diffpy/diffpy.utils/issues/156
- [x] https://github.com/diffpy/diffpy.utils/issues/139

@Sparks29032 

- [x] https://github.com/diffpy/diffpy.utils/issues/145

More new added for 3.6.0
- All other remaining issues

### 3.5.1 Archive

To be closed for 3.5.1 by @alisnwu
https://github.com/diffpy/diffpy.utils/milestone/2

Overall communication channel for 3.5.1 and we can set priorities here for communicating with all members in the group:

The following are already resolved - but will be closed once pdfmorph confirms it (after 11/3)
- [x] https://github.com/diffpy/diffpy.utils/issues/132
- [x] https://github.com/diffpy/diffpy.utils/issues/131

The following are what we need by next week:
- [x] https://github.com/diffpy/diffpy.utils/issues/140
- [x] https://github.com/diffpy/diffpy.utils/issues/138
- [x] https://github.com/diffpy/diffpy.utils/issues/137
- [x] https://github.com/diffpy/diffpy.utils/issues/129
- [x] https://github.com/diffpy/diffpy.utils/issues/144

**Comment by alisnwu** [comment]
@bobleesj I think everything is good so far! I made edits to #140 and #129 based on simon's edits and #137 should be done after #138 is decided. I believe #144 is just waiting to be merged

**Comment by bobleesj** [comment]
@alisnwu Would it be possible to do the 3.5.1 checklist tomorrow? I think https://github.com/diffpy/diffpy.utils/pull/158 needs to be into a new PR as requested. Thanks a lot on your hard work.

**Comment by alisnwu** [comment]
Yep sounds good, I will make a new PR for that tonight

**Comment by bobleesj** [comment]
@alisnwu almost there! great progress.

**Comment by sbillinge** [comment]
@alisnwu I merged the other PR that previously had all the function name changes.

We will need a new PR where we deprecate `Diffraction_object` in favor of `DiffractionObject`.

Let's do that one as a practice.  We can tackle `loadData` later if we want to, so for now just leave that as it is.  You would normally deprecate something for two mid-level release numbers at least.

**Comment by bobleesj** [comment]
https://github.com/diffpy/diffpy.utils/pull/165

@alisnwu Just to make sure, the deprecation task is handled in the PR above to to set a standard where we can build upon.

**Comment by bobleesj** [comment]
@alisnwu Would you make a PR for the one remaining issue above?

**Comment by bobleesj** [comment]
@alisnwu Alison, I've noticed this PR is created https://github.com/diffpy/diffpy.utils/pull/169 by @Sparks29032 

What's going on? Does it close #137 ?

**Comment by alisnwu** [comment]
@bobleesj Andrew was not experiencing the errors from sphinx that I was on his local, so this would PR should close #137

**Comment by bobleesj** [comment]
Will you be making the release checklist, @alisnwu? Ideally, we need some buffer time before tomorrow's group meeting to test the package (Please see the checklist). We should probably do it this morning or early afternoon latest to make it realistic, since @sbillinge needs to release packages manually.

**Comment by alisnwu** [comment]
@bobleesj Yep, already made the checklist. However, still need to check off the new tasks in the 3.5.1 milestone

**Comment by bobleesj** [comment]
@sbillinge just to make sure - 3.5.1 is now modified to 3.6.0 release?

**Comment by bobleesj** [comment]
> yes, I added scope, some of which is API changing, so I bumped it to 3.6.0 and changed the date to the end of December.
> 
> 
> 
> If (as a team) we give this some focus, I think we can do it and have a nice public release.  tbh, we need a manuscript that describes the DiffractionObjects....

@yucongalicechen Please keep us posted regarding this!

**Comment by yucongalicechen** [comment]
> > yes, I added scope, some of which is API changing, so I bumped it to 3.6.0 and changed the date to the end of December.
> > If (as a team) we give this some focus, I think we can do it and have a nice public release.  tbh, we need a manuscript that describes the DiffractionObjects....
> 
> @yucongalicechen Please keep us posted regarding this!

Thanks Bob. @sbillinge not sure I'm the best person for this task, but I can help!

**Comment by sbillinge** [comment]
@yucongalicechen I think @alisnwu is in charge of `diffpy.utils` so she would shephard this home, but she is not expected to do everything, but assign tasks around to people.  I can help with that if @alisnwu can then keep things on track to ensure the release succeeds.

**Comment by bobleesj** [comment]
@alisnwu @yucongalicechen Based on Monday's call, I've organized the 3.6.0 release items above.  
 Could you please confirm whether the issues are assigned as expected?

**Comment by bobleesj** [comment]
@alisnwu, what issues will you be addressing by next week? @yucongalicechen mentioned that if your hands are currently tied, she could help resolve your issues during the remainder of the semester. Please let us know so that we can maintain the momentum as a team here.

**Comment by alisnwu** [comment]
@bobleesj Hey Bob, I'll be able to work on the issues a lot today so hopefully will be making good progress on the issues. May not be able to write all of the tests in time because I have 3 more midterms next week so may need help on some of the tests @yucongalicechen Thank you so much for offering to help out!

**Comment by bobleesj** [comment]
@alisnwu Look forward to seeing good progress. Please report back by the end of the day on which tests/issues you need help with so that we can move tasks around.

**Comment by sbillinge** [comment]
@bobleesj #195 is on milestone 3.7.0 release, so I tihnk we assign it to a newbie who can take their time on it.  Maybe remove it from this PR which is a 3.6.0 release one.

**Comment by bobleesj** [comment]
@sbillinge https://github.com/diffpy/diffpy.utils/issues/201 I can take care of this issue for 3.6.0 release?

**Comment by sbillinge** [comment]
Shall we use this opportunity to change `input_scattering_quantity` to `input_data`? Shorter and easier to remember? Thoughts?

**Comment by bobleesj** [comment]
@sbillinge 
yes, I like it. Considering this is the only function that is used within `init`, I think making it shorter does not confuse for users and future devs (also the doc string is pretty straight forward `insert a new scattering quantity into the scattering object`)

**Comment by bobleesj** [comment]
Mid-term progress report (Dec 18, 2024)

Remaining tasks with responsibility:

- Bob (@bobleesj) leads docstrings - use numpy standard and set BG group's way with @sbillinge 
- Yucong (@alisnwu) leads official documentation
- Alison and Bob (@alisnwu and @bobleesj) write remaining tests for `DiffractionObject`

Good progress made!

**Comment by sbillinge** [comment]
closing as comleted.  @bobleesj there is an unchecked todo in a comment above in case you want to harvest it and do something with it, but I want to close this to finalize the milestone. 

- [ ] Bob organizes a group hackathon for improvement docstrings in diffpy.utils for training and learning.

**Comment by bobleesj** [comment]
@sbillinge yup, didn't get to organize a quick hackathon but @yucongalicechen and I were able to resolve docstrings. 

Early in the semester, I could organize a quick rundown on our group's standards with examples available through our official cookiecutter doc.
